src/main/java/com/r307/arbitrader/service/ticker/TickerStrategyProvider.java

```diff
@@ -0,0 +1,39 @@
+package com.r307.arbitrader.service.ticker;
+
+import com.r307.arbitrader.config.NotificationConfiguration;
+import com.r307.arbitrader.service.ErrorCollectorService;
+import com.r307.arbitrader.service.ExchangeService;
+import com.r307.arbitrader.service.event.StreamingTickerEventPublisher;
+import org.springframework.stereotype.Component;
+
+import javax.inject.Inject;
+
+@Component
+public class TickerStrategyProvider {
+
+    private final ErrorCollectorService errorCollectorService;
+    private final StreamingTickerEventPublisher streamingTickerEventPublisher;
+    private final NotificationConfiguration notificationConfiguration;
+
+    @Inject
+    public TickerStrategyProvider(ErrorCollectorService errorCollectorService,
+                                  StreamingTickerEventPublisher streamingTickerEventPublisher,
+                                  NotificationConfiguration notificationConfiguration) {
+
+        this.errorCollectorService = errorCollectorService;
+        this.streamingTickerEventPublisher = streamingTickerEventPublisher;
+        this.notificationConfiguration = notificationConfiguration;
+    }
+
+    public TickerStrategy getStreamingTickerStrategy(ExchangeService exchangeService) {
+        return new StreamingTickerStrategy(errorCollectorService, exchangeService, streamingTickerEventPublisher);
```

Comment:
I'm not sure it matters, but these `TickerStrategy` objects were Spring beans before, so there was only a single instance of each. This will return a new instance each time which is a change. I believe they are all stateless (necessary because they were shared) so this shouldn't break anything but it may result in allocating more objects.

Reply:
I don't think it should be a serious issue, unless the user is using a lot of exchanges. If you agree, I would keep it this way and maybe in the future we can improve it.
Saying this because, from the top of my mind, I think it will be tricky to improve it due to the circular dependencies. 

`ExchangeService` depends `TickerStrategy`. On the other hand `TickerStrategy` methods depend on `ExchangeService`. I thought about creating at startup an instance of each `TickerStrategy` implementations. Then when a call to `getStreamingTickerStrategy` or `getParallelTickerStrategy` or `getSingleCallTickerStrategy` was made it would return the appropriated (and already created) instance of `TickerStrategy` implementation but before returning it would need to set the exchange obect into `TickerStrategy` (due to the dependency `TickerStrategy` implementation dependency on `Exchange`) which would force us to remove the `final` keyword on the `Exchange` field of each `TickerStrategy` implementation.

Was I clear? :sweat_smile:

Reply:
Ok, yes. I don't think it will be a problem. I double checked and we're only creating a new instance when the exchange is set up, not for every tick. That should be fine.